app/models/conference.rb

```diff
@@ -317,6 +317,60 @@ def event_distribution
   end
 
   ##
+  # Returns a hash with scheduled vs unscheduled events
+  # { "Scheduled" => { value: number of confirmed and scheduled events, color: color },
+  #   "Unscheduled" => { value: number of confirmed and unscheduled events, color: color }
+  #
+  # ====Returns
+  # * +hash+ -> hash
+  def scheduled_event_distribution
+    confirmed_events = program.events.where(state: 'confirmed')
+    scheduled_value =  { 'value' => confirmed_events.where('start_time IS NOT NULL').count, 'color' => 'green' }
+    unscheduled_value =  { 'value' => confirmed_events.where('start_time IS NULL').count, 'color' => 'red' }
+    { 'Scheduled' => scheduled_value, 'Unscheduled' => unscheduled_value }
+  end
+
+  ##
+  # Returns a hash with Registration attended vs. Registration not attended
+  # { "Attended" => { value: number of registration attended, color: color },
+  #   "Not attended" => { value: number of registration not attended, color: color }
+  #
+  # ====Returns
+  # * +hash+ -> hash
+  def registration_distribution
+    reg = registrations.includes(:user)
+    attended_value =  { 'value' => reg.where(attended: true).count, 'color' => 'magenta' }
+    not_attended_value =  { 'value' => reg.where.not(attended: true).count, 'color' => 'blue' }
+    { 'Attended' => attended_value, 'Not attended' => not_attended_value }
+  end
+
+  ##
+  # Returns a hash with affiliation =>
+  # {value: count of registration whose user has that affilation, color: color}
+  # In case that the affiliation is nil, it groups it in not_attended_value
+  #
+  # ====Returns
+  # * +hash+ -> hash
+  def affiliation_distribution
+    affiliations = []
+    registrations.each do |r|
+      affiliations << (r.user.affiliation ? r.user.affiliation : 'None')
+    end
+    counted_affiliations = affiliations.group_by(&:capitalize).map {|k, v| [k, v.length]}
```

Comment:
Hm, I'd prefer to do this in some ActiveRecord grouping? Because using database features is way faster than doing it in Ruby code.